ovirt_imageio/_internal/qemu_nbd.py

```diff
@@ -147,7 +152,10 @@ def start(self):
         # Build a 'json:{...}' filename allowing control all aspects of the
         # image.
 
-        file = {"driver": "file", "filename": self.image}
+        if self._is_block_dev():
+            file = {"driver": "host_device", "filename": self.image}
+        else:
+            file = {"driver": "file", "filename": self.image}
```

Comment:
This can be simpler:

    driver = "host_device" if self._is_block_dev() else "file"
    file = {"driver": driver, "filename": self.image}

But calling os.stat() multiple times does not make sense. We should really call it once when starting and set a flag.